tests/draft2020-12/unevaluatedItems.json

```diff
@@ -433,5 +433,134 @@
                 "valid": false
             }
         ]
+    },
+    {
+        "description": "item is evaluated in a sibling schema to unevaluatedItems",
+        "schema": {
+            "type": "array",
+            "prefixItems": [
+                {
+                    "type": "string"
+                }
+            ],
+            "unevaluatedItems": false,
+            "anyOf": [
+                {
+                    "prefixItems": [
+                        {
+                            "type": "string"
+                        }
+                    ]
+                }
+            ]
+        },
+        "tests": [
+            {
+                "description": "no extra items",
+                "data": [
+                    "test"
+                ],
+                "valid": true
+            },
+            {
+                "description": "uncle keyword evaluation is not significant",
+                "data": [
+                    "test",
+                    "test"
+                ],
+                "valid": false
+            }
+        ]
+    },
+    {
+        "description": "item is evaluated in an uncle schema to unevaluatedItems",
+        "schema": {
+            "type": "object",
+            "properties": {
+                "foo": {
+                    "type": "array",
+                    "prefixItems": [
+                        {
+                            "type": "string"
+                        }
+                    ],
+                    "unevaluatedItems": false
+                  }
+            },
+            "anyOf": [
+                {
+                    "properties": {
+                        "foo": {
+                            "prefixItems": [
+                                true,
+                                {
+                                    "type": "string"
+                                }
+                            ]
+                        }
+                    }
+                }
+            ]
+        },
+        "tests": [
+            {
+                "description": "no extra items",
+                "data": {
+                    "foo": [
+                        "test"
+                    ]
+                },
+                "valid": true
+            },
+            {
+                "description": "uncle keyword evaluation is not significant",
+                "data": {
+                    "foo": [
+                        "test",
+                        "test"
+                    ]
+                },
+                "valid": false
+            }
+        ]
+    },
```

Comment:
Similar to what I mentioned in another comment for `unevaluatedProperties`, I think this can be simplified and still test the same thing.

```suggestion
    {
        "description": "item is evaluated in an uncle schema to unevaluatedItems",
        "schema": {
            "type": "object",
            "allOf": [
                {
                    "type": "array",
                    "prefixItems": [
                        { "type": "string" }
                    ],
                    "unevaluatedItems": false
                },
                {
                    "properties": {
                        "foo": {
                            "prefixItems": [
                                true,
                                { "type": "string" }
                            ]
                        }
                    }
                }
            ]
        },
        "tests": [
            {
                "description": "no extra items",
                "data": {
                    "foo": [
                        "test"
                    ]
                },
                "valid": true
            },
            {
                "description": "uncle keyword evaluation is not significant",
                "data": {
                    "foo": [
                        "test",
                        "test"
                    ]
                },
                "valid": false
            }
        ]
    },
```

Reply:
Similar to my comment above on the simplification of the unevaluatedProperties test: this appears to be basically a variant of "unevaluatedItems can't see inside cousins" on line 418. The evaluated and unevaluated items have the same parent keyword, though structurally not directly under the parent. I think the idea is to test that unevaluatedItems is not influenced by evaluation of items in a different keyword branch of the schema tree.

Reply:
I agree with @jdesrosiers that we only need one test per keyword and that this one is the most appropriate. However, I don't think it should be simplified.

Also, not too keen on the term 'uncle' - it's a little confusing and potentially misleading!

Reply:
as with unevaluatedProperties, the use of different keywords can tickle different behaviour in the evaluator.